### DISCUSSION TITLE
Unique violation for non-existent records invalidly persisting across insert transactions since update from 1.4.8 to 2.0.4, MySQL

### DISCUSSION THREAD

**Discussion by biot023** [diesel-rs/diesel#3928] [Q&A]
Hello -- I've been upgrading our project from 1.4.8 to 2.0.4 and have come across a weird error.
I have two requests in acceptance tests, each of which fails if the first is run before it. The way it fails is the second test performs an insert that generates a uniqueness violation error that quotes the data from the first test in its error message.
I think maybe the previous error in the MySQL engine isn't being cleared or something?
The uniqueness key is on two fields (call them `field_a` and `field_b`). Here is the order of everything that is happening in the tests:

TEST 1
- Clear all DB records in test setup
- Create a record with values 100 and 200 (for example) for `field_a` and `field_b` -- this passes

TEST 2
- Clear all DB records in test setup
- Create a record with values 300 and 400 for `field_a` and `field_b` -- this fails with a uniqueness violation that quotes the inital "TEST 1" values of 100 and 200

So, as you see, the operation not only wasn't trying to insert the older values but *those records no longer even exist*.

I haven't been able to re-create a simple, self-contained version of this error, sorry. That said, I can't see how it could be other than a library-level issue.
My apologies if that is myopic of me. :)

I *have* debugged by logging the values all the way down to outputting the insert SQL in the `diesel`-level insert code and that generates the expected SQL to insert the values 300 and 400 to `field_a` and `field_b` -- then promptly fails with the uniqueness violation that quotes the previous, now-non-existent values of 100 and 200.

For full information, the acceptance tests are run in Ruby/Cucumber using ActiveRecord to set up, and the `diesel` code is being called by an `async-graphql` client, but as I said, I've debugged the issue all the way back to the query being called and that outputs as it should. The DB engine is MySQL.

If anybody could offer any insight into how to proceed to solve this I would be *incredibly* grateful -- currently I'm looking at just abandoning the update work as I've no idea how to proceed beyond this point.

Thank you,
   Doug.

**Comment by weiznich** [comment]
It's hard to help without having a reproducible example. What I can say is that we are currently not aware of such an issue.

I would likely enable query logging in mysql side and verify that what is actually send to the database matches what's expected there: https://dev.mysql.com/doc/refman/8.0/en/query-log.html

Using that information it hopefully shouldn't be too hard to create a reproducible example.

**Reply by biot023** [reply]
Brilliant, thank you -- I'll get on that, this afternoon.
